Structure

dodoburner opened the pull request:
Here is a [video preview](https://loom.com/share/2db46e1b81c14d38a93355b2e7718548) of the app
In this pull request I have created the following features for this app:

- The app is setup with React & Redux
- Each page has a slice created for storing the data (continents, countries, cities)
- The app fetches the data needed for the page when the page is opened and stores it in the store using asyncThunks
- 2 APIs are used simultaneously: 
One is for getting all countries for a given continent and the top 20 biggest cities for a country.
The second is for fetching all countries that have air pollution data and for getting air pollution data for a given city
- The home page has a search bar that fetches data for a city and displays the city page 
- If the city isn't available or it is typed in incorrectly, an error page is displayed

EricMbouwe commented on src/App.js:
```diff
@@ -1,23 +1,22 @@
+          <Route path="" element={<Home />} />
+          <Route path="/continent" element={<Continent />} />
+          <Route path="/country" element={<Country />} />
+          <Route path="/city" element={<City />} />
```
- [ ] There is no direct connection between your routes. For example, if I go to a country from the URL, I should be able to go to any city of that country by using a city parameter.

To fix that you can use this syntax
```js
<Route path="/country/:cityId" component={City} />
```

EricMbouwe commented on src/components/City.js:
```diff
@@ -0,0 +1,148 @@
+import '../styles/City.css';
+
+export default function City() {
+  const location = useLocation();
```
- [ ] I would recommend using [URL parameters](https://v5.reactrouter.com/web/example/url-params) to handle data from a specific item.

You would get something like this:

```js
import { useParams } from "react-router-dom";

const { cityId } = useParams();
```

EricMbouwe commented on src/components/Country.js:
```diff
@@ -0,0 +1,52 @@
+export default function Country() {
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const { id, name } = location.state;
+
```
- [ ] As mentioned above I would recommend using [URL parameters](https://v5.reactrouter.com/web/example/url-params) to handle data from a specific item.

Here is the error that occurs when we try to access the country route directly from the browser URL. `{ id, name }` doesn't exist as it is null
<img width="355" alt="image" src="https://user-images.githubusercontent.com/40503704/187992627-e9169598-dfaf-4287-a84b-4dec3f90fa12.png">

EricMbouwe reviewed (changes requested):
Hi @dodoburner, 

Good job so far! 🔥
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes ♻
- [ ]  You should deploy the application using Heroku, Netlify or GitHub Pages. Let's fix that
- [ ] Let's add a link to the deployed project in the README
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please check the sound of the presentation video and fix it. Currently, there is no sound ♻

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

dodoburner commented:
@EricMbouwe Thank you for your observations. I have now refactored the code to use useParams(). Regarding the deployment, I will deploy once the project is approved and finished. The video preview is also updated.

codepantha commented on README.md:
```diff
@@ -1,70 +1,100 @@
+  - [x] React
+  - [x] Redux
+
+## Live Demo Link
+
+Coming soon...
+
```
- [ ] Please endeavour to deploy the application using Heroku, Netlify or GitHub Page. You need to do this in order to get approved 😎. 
- [ ] Once you've deployed the application, kindly add the live link to your README. This is important so that people who are perusing your README and quickly take a look at the live app and see what your project is all about 👍

codepantha reviewed (changes requested):
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @dodoburner 👋, 

Well done implementing most of the required changes 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!
<img src="https://media1.giphy.com/media/YT95XJOLvY1t2SJgpR/giphy.gif"/>

## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Used correct GitFlow
✔️ Beautiful UI design

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BenMukebo commented on README.md:
```diff
@@ -1,70 +1,100 @@
+  - [x] React
+  - [x] Redux
+
+## Live Demo Link
+
+[Live Demo](https://dodoburner.github.io/europe-air-pollution/)
```
- [ ] Great job with the video presentation 📸 👍🏻  however as requested by the previous reviewer, make sure the project is deployed properly, and the home and detail page works as expected. Your deployed link in the README directs me to an empty page. Kindly make sure the project is deployed properly using Heroku, Netlify, or GitHub Pages. 👍🏻 💯

BenMukebo reviewed (changes requested):
## Hi @dodoburner,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉

- Professional README file. :heavy_check_mark:
- Pull Request has a good summary. ✔️
- Good presentation video. ✔️ 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mikethreels reviewed (approved):
Hi Dorian, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job implementing the changes requested by the previous CR

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
